### PR TITLE
Activate wallet-connect deeplink with qr-mode

### PR DESCRIPTION
**Pull request by henrypalacios**
closes: #562 

Activate walletconnect deeplink.

:warning: **Problems related to this PR:** 
- Functionality Issue: #671
- UI issues: #670, #659

### CODE REVIEW

File: src/theme/globalStyle.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ export const GlobalStyle = createGlobalStyle<{ theme: any }>`
 
 export const ThemedGlobalStyle = createGlobalStyle`
   html {
-    color: ${({ theme }) => theme.text1};
+    color: ${({ theme }) => theme.text4};
```
Comment (henrypalacios) [inline src/theme/globalStyle.tsx:29]: walletconnect modal take the `body>color` style, I didn't find another way to change the `text color`, and it does not seem to affect anything, what do you think @alongoni?

**Review by josojo — APPROVED** [review]
Nice, really like tthis PR. Big improvement.

Well done!

**Comment by elena-zh** [comment]
Hey @henrypalacios , @josojo 
I have verified the issue. Great job! However, here are several notes: 
1. I have not been able to connect the app with the Safe app. See the video: 
https://drive.google.com/file/d/1LK8K4uXiIvfyYoouOYWwkpjlXOA7HXh_/view

2. Sometimes the link is not copied: I see the message that it was copied, but when I paste it, I see the previously copied value
https://drive.google.com/file/d/1aurDHyED-B2q2gyYkhXpenaTFotmi0bN/view

3. Cross icon in the WalletConnect modal has a double line
cross with double line

4. These issues might not be related to our current implementation but to the integrated component, however, I have to mention that: 
4.1. 404 page is opened when call connection in iOS using
 - Crypto.com
 -  1inch
 - CYBAVO
 - AlphaWallet
 - EasyPocket
 - SparkPoint
 - PEAKDEFI
 - Aktionariat
 - Flare
 - AToken
 - Tongue
 - RWallet
 - PlasmaPay
 - O3Wallet
 - Defiant
4.2. Nothing happens, when the next apps are not installed in an iOS mobile device:
 - Pillar
 - Argent
 - ImToken
 - TokenPocket
 - Ledger
 - Dharma
 - Huobi
 - HaloDefi

5. I'm not able to see app's list in my Android device: 
image

Could you please take a look into these issues?

Thanks

**Comment by elena-zh** [comment]
Hi @henrypalacios , great job!
 - Cases 1-2 are fixed. 

 - I have created a separate issue for the case 3 - #659 and assigned @alongoni to it. 

 - As for the cases 4 and 5. In Android it works fine: the app suggests a user to open any supported app which can be connected using WalletConnect option. 
However, the issue I mentioned is still reproducible on **iOS** devices. 
@josojo , could you please take a look at the case 4 in my [previous comment ](https://github.com/gnosis/ido-ux/pull/650#issuecomment-855770392) and clarify whether we should leave it as it is?
The issue is that for several apps nothing happens when a user selects them in case if they are not installed to the device. Another case that a lot of apps navigate to the 404 page when select them. 

 - Also, I have faced another issue that is related to this pr: walletconnect modal appears after disconnection of a wallet. And it is no closed when connect a wallet using it once again. @henrypalacios , could you please take a look at it? 
https://drive.google.com/file/d/1POb25DYAo20pKUplDAEVJmkVWA8uuY82/view

 - BTW, we have an low priority issue #599 on the board related to a modal header name. Could you please fix it in this pr?

Thanks

**Comment by josojo** [comment]
> * @josojo , could you please take a look at the case 4 in my [previous comment ](https://github.com/gnosis/ido-ux/pull/650#issuecomment-855770392) and clarify whether we should leave it as it is?
>   The issue is that for several apps nothing happens when a user selects them in case if they are not installed to the device. Another case that a lot of apps navigate to the 404 page when select them.

Yes, this is fine, let's not change anything

**Comment by ramirotw** [comment]
> * Also, I have faced another issue that is related to this pr: walletconnect modal appears after disconnection of a wallet. And it is not closed when connect a wallet using it once again. @henrypalacios , could you please take a look at it?
>   https://drive.google.com/file/d/1POb25DYAo20pKUplDAEVJmkVWA8uuY82/view

Might be related to https://github.com/gnosis/ido-ux/issues/600

**Comment by ramirotw** [comment]
Also, on a HD screen both the modal's header and the copy tooltip are not 100% visible

image

**Comment by henrypalacios** [comment]
@elena-zh continuing your notes:
--
>Also, I have faced another issue that is related to this pr: walletconnect modal appears after disconnection of a wallet. And it is not closed when connect a wallet using it once again. @henrypalacios , could you please take a look at it?   

I have not been able to find a way to solve this, the problem is associated when the **walletconnect** modal appears immediately after disconnecting. The strange thing is that at times it works and at other times it does not. [As in this video that occurs after the third attemp](https://drive.google.com/file/d/1CKeD9YIj9JBaVbn6OSclzOv7xlo7Cpjg/view?usp=sharing)

> Could you please fix it in this pr?

closes #599 also.

@ramirotw 
--
> Might be related to #600

@ramirotw It doesn't look like it because when refreshing walletconnect it is not really there (it is the `InjectedConnector` that remains).

> Also, on a HD screen both the modal's header and the copy tooltip are not 100% visible.

Do you think we should open another issue for that, since I'm not sure if it's possible [to just make a wrapper](https://github.com/gnosis/ido-ux/blob/ed1f1a9a893af140494f8d3e4df746ef39e1dc5e/src/connectors/index.ts#L51)?

**Comment by elena-zh** [comment]
Hi @henrypalacios , 
#599 Looks good to me now!

However, today I have not been able to connect a wallet using walletconnect option: each time when I select this option, I get the app crash =(
image

As for the mentioned above issues, I have created tickets for them:
#670 - truncated modal header
#671 - Walletconnect modal appears after a wallet disconnection

Thanks

**Comment by henrypalacios** [comment]
closes #671

**Comment by josojo** [comment]
@henrypalacios can you give a summary on what is working and what is not working? 

Do you need help with anything particular? It would be great if we can get it ready for release.

**Comment by elena-zh** [comment]
@henrypalacios , @josojo , 
I have retested the PR. Here are my notes: 
1. #671 is still reproducible: connection window may appear again (50/50%) when I press on the 'Disconnect' button. I think, It worth fixing before the release due to connection does not work as expected when reconnect a wallet from the newly appeared modal window (see the issue description)
2. Connection with Gnosis Safe works fine. I was able to run place/cancel/claim order transactions from safe, however, transaction status is not tracked: all transactions are in 'Pending' status in the GA app even if they are successfully completed: #675. I think, we should fix it before the release. 
I have to notice, that transactions appropriately change their status when use walletconnect via barcode. 
3. still, there are several UI issues unfixed (#670, #659 ), but they are not high priority ones and can be fixed later.

Thanks

**Comment by elena-zh** [comment]
Hi @henrypalacios , @hpmaxi , 
the #671 issue seems to be fixed now. 
As for the pending transactions issue, I cannot say that it is fixed. Now all 'pending' transactions do not appear at all: when I place an order, it is not displayed in the 'Your orders' section with 'pending' state. 
Also, no confirmation pop-ups when a transaction ends. Then, nothing is displayed in the 'Your transactions'  list even after reloading the app
https://drive.google.com/file/d/1alYBP2fdT47AqVqLLntj_n0bq2MRBtES/view

**Comment by hpmaxi** [comment]
@elena-zh @josojo 
I think this could be closed and make another issue with the safe related issues of the Elena's last comment. And close #671 too.

About safe, as far as I know, we are not updating the transactions state with the internal calls made by the safe, so maybe fixing that is completely unrelated to qr-mode.

**Comment by elena-zh** [comment]
Hi @hpmaxi , here a separate issue for Gnosis Safe transactions #693. 
I was also able to reproduce #671 issue again - a reopened it and put it into the Review state. If we take a decision to fix it in another branch, it would be nice to unlink it from this PR and then move it into another column on the board. 

Thanks

**Comment by josojo** [comment]
Yes, I think its fine to track them with issues and merge what we have already.

**Comment by henrypalacios** [comment]
In this PR try to handle the error when the[ app is crashed when connect a wallet with another unsupported network](https://github.com/gnosis/ido-ux/issues/658)

**Comment by elena-zh** [comment]
Hi @henrypalacios , thank you for the fix!
Now the app is not crashed, however, works a bit strange. Look:
1. When I connect to another supported network, I receive an error message to change a network.


2. A lot of errors in console appear right after connection
here is the video: ttps://drive.google.com/file/d/10AGylZilk3K166nEHCnK2JbrZl_ae57r/view

It would be great to use the same flow as it works for Metamask connection: 
 - show a connected state with a message to select another network instead of the 'Connect wallet'  button when a user is connected to a supported network
 image
 - show an error and do not connect when a user is not in supported network. 
Metamask flow: https://drive.google.com/file/d/1eORx4AoXUnaavA-GPcuEIFyQaIN0kYY_/view

**Comment by henrypalacios** [comment]
Ok @elena-zh , I'm going to revert the changes related with #658 by merged this PR, and I will continue it there.

**Pull request merged**